javascript/features/limits/limits_decider.js

```diff
@@ -150,6 +154,15 @@ export class LimitsDecider extends PlayerEventObserver {
         return null;
     }
 
+   // Processes if the player is present in vehicle and makes sure if the player isn't in one.
+   // This proces is required to avoid uncertain bugs .
+    processVehicleRequirement(player) {
+        if(player.state == Player.kStateVehicleDriver && Player.kStateVehiclePassenger)
```

Comment:
I don't think this if statement is doing what you want.

Currently it states: If the player's state equals to `Player.kStateVehicledriver` AND the value of `Player.kStateVehiclePassenger` is truthy.

The latter is _always_ true because it's a static value. I assume you want a || `player.state === Player.kStateVehiclePassenger`

Also please test this as passenger. 
Also please try to use `===` instead of `==`.